### ISSUE TITLE
[BUG] Armeria exception when under pressure

### ISSUE THREAD

**Issue by dlvenable** [opensearch-project/data-prepper#2206]
**Describe the bug**

When sending large volumes of trace data to Data Prepper, the Data Prepper process has exceptions which are being thrown from the Armeria framework.

```
2023-01-27T03:11:04,274 [armeria-common-worker-epoll-3-1] WARN  io.netty.util.concurrent.AbstractEventExecutor - A task raised an exception. Task: com.linecorp.armeria.common.RequestContext$$Lambda$1600/0x00000008015ca908@2d24213e
java.lang.NullPointerException: null
```

(The full stack trace is below)

**To Reproduce**

1. Configure Data Prepper with a trace pipeline configure with too large of a buffer. It needs to include stateful processing. I used the following.

```
entry-pipeline:
  buffer:
    bounded_blocking:
      buffer_size: 20000000
      batch_size: 4000
  source:
    otel_trace_source:
      ssl: true
      sslKeyCertChainFile: "config/default_certificate.pem"
      sslKeyFile: "config/default_private_key.pem"
      port: 21890
      unframed_requests: true
  sink:
    - pipeline:
        name: "raw-pipeline"
    - pipeline:
        name: "service-map-pipeline"

raw-pipeline:
  buffer:
    bounded_blocking:
      buffer_size: 200000
      batch_size: 4000
  source:
    pipeline:
      name: "entry-pipeline"
  processor:
    - otel_trace_raw:
        trace_flush_interval: 1
  sink:
    - opensearch:
        hosts: [ "https://my-aos-domain.es.amazonaws.com" ]
        aws_sts_role_arn: "arn:aws:iam::123456789012:role/MyRole"
        aws_sigv4: true
        bulk_size: 20
        index_type: trace-analytics-raw

service-map-pipeline:
  delay: "100"
  buffer:
    bounded_blocking:
      buffer_size: 200000
      batch_size: 4000
  source:
    pipeline:
      name: "entry-pipeline"
  processor:
    - service_map_stateful:
        window_duration: 10
  sink:
    - opensearch:
        hosts: [ "https://my-aos-domain.es.amazonaws.com" ]
        aws_sts_role_arn: "arn:aws:iam::123456789012:role/MyRole"
        aws_sigv4: true
        bulk_size: 20
        index_type: trace-analytics-service-map
```

2. Run Data Prepper. I ran it in the background on an EC2 instance.

```
nohup env JAVA_OPTS='-Xmx7g -Xms7g bin/data-prepper &
```

3. Send a large volume of trace data. I used the tracegen tool from the Open Telemetry collector.

4. After a short period of time, Data Prepper threw the following exception almost several dozen times.

```
2023-01-27T03:11:04,274 [armeria-common-worker-epoll-3-1] WARN  io.netty.util.concurrent.AbstractEventExecutor - A task raised an exception. Task: com.linecorp.armeria.common.RequestContext$$Lambda$1600/0x00000008015ca908@2d24213e
java.lang.NullPointerException: null
    at com.linecorp.armeria.internal.shaded.guava.base.Preconditions.checkNotNull(Preconditions.java:889) ~[armeria-1.20.3.jar:?]
    at com.linecorp.armeria.internal.shaded.guava.collect.SingletonImmutableList.<init>(SingletonImmutableList.java:39) ~[armeria-1.20.3.jar:?]
    at com.linecorp.armeria.internal.shaded.guava.collect.ImmutableList.of(ImmutableList.java:100) ~[armeria-1.20.3.jar:?]
    at com.linecorp.armeria.internal.common.stream.OneElementFixedStreamMessage.drainAll(OneElementFixedStreamMessage.java:61) ~[armeria-1.20.3.jar:?]
    at com.linecorp.armeria.internal.common.stream.FixedStreamMessage.collect(FixedStreamMessage.java:201) ~[armeria-1.20.3.jar:?]
    at com.linecorp.armeria.internal.common.stream.FixedStreamMessage.lambda$collect$2(FixedStreamMessage.java:188) ~[armeria-1.20.3.jar:?]
    at com.linecorp.armeria.common.RequestContext.lambda$makeContextAware$3(RequestContext.java:566) ~[armeria-1.20.3.jar:?]
    at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.86.Final.jar:4.1.86.Final]
    at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.86.Final.jar:4.1.86.Final]
    at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.86.Final.jar:4.1.86.Final]
    at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403) ~[netty-transport-classes-epoll-4.1.86.Final.jar:4.1.86.Final]
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.86.Final.jar:4.1.86.Final]
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.86.Final.jar:4.1.86.Final]
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.86.Final.jar:4.1.86.Final]
    at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

**Expected behavior**

I'd expect that Data Prepper gives a clearer exception when under high pressure. Additionally, the exceptions should not be coming from the Armeria framework.

**Environment (please complete the following information):**
 - OS: Amazon Linux 2

**Additional context**

This is on the `main` branch of Data Prepper, not a build.

Also, I can avoid this error by enabling the circuit breaker.

**Comment by dlvenable** [comment]
This may be caused by a bug in Armeria which is scheduled to be fixed in 1.22.0.

https://github.com/line/armeria/pull/4652

**Comment by dlvenable** [comment]
The above fix worked, but then another issue was manifest. This issue appears to be resolved in https://github.com/line/armeria/pull/4667.

This issue should be completed after updating to use the latest release of Armeria when it is ready.

**Comment by dlvenable** [comment]
Resolved by #2274.
